sql/sql_delete.cc

```diff
@@ -869,6 +869,47 @@ int Query_result_delete::prepare(List<Item> &values, SELECT_LEX_UNIT *u)
 }
 
 
+/**
+  Test that the subject table (of DELETE) has a cascade foreign key
+  parent present in the query.
+
+  @param  thd        thread handle
+  @param  table      table to be checked (must be updatable base table)
+  @param  table_list List of tables to check against
+
+  @retval bool       True if cascade parent found.
+*/
+static bool has_cascade_dependency(THD *thd, const TABLE_LIST &table,
+                                   TABLE_LIST *table_list)
+{
+  DBUG_ASSERT(&table == const_cast<TABLE_LIST &>(table).updatable_base_table());
+
+  List <st_handler_tablename> fk_table_list;
+  List_iterator<st_handler_tablename> fk_table_list_it(fk_table_list);
+
+  table.table->file->get_cascade_foreign_key_table_list(thd, &fk_table_list);
+
+  st_handler_tablename *tbl_name;
+  while ((tbl_name= fk_table_list_it++))
+  {
+    for (TABLE_LIST* curr= table_list; curr; curr= curr->next_local)
+    {
+      const bool same_table_name= lower_case_table_names ?
```

Comment:
IMHO this is now asking for bool db_or_table_name_equal(const char *, const char *) helper (it's fascinating that server code does not have it already)

Reply:
It is, but it would be either 4 string parameter, or many overloads (TABLE, TABLE_LIST, st_handler_table_name all combinations, for example...)

Reply:
I had something in mind with the above signature: 
bool db_or_table_name_equal(const char *, const char *);
which compares only two given DB or table names
Perhaps the "db_or_table" name part can be improved

Reply:
That's a good idea, I thought you meant "and" there.